File: TransCoder_tokenized_test_set_functions/cpp/COUNT_INDEX_PAIRS_EQUAL_ELEMENTS_ARRAY_1.cpp
int countPairs ( int arr [ ], int n ) {
  unordered_map < int, int > mp;
  for ( int i = 0;
  i < n;
  i ++ ) mp [ arr [ i ] ] ++;
  int ans = 0;
  for ( auto it = mp . begin ( );
  it != mp . end ( );
  it ++ ) {
    int count = it -> second;
    ans += ( count * ( count - 1 ) ) / 2;
  }
  return ans;
}

File: TransCoder_tokenized_test_set_functions/java/COUNT_INDEX_PAIRS_EQUAL_ELEMENTS_ARRAY_1.java
public static int countPairs ( int arr [ ], int n ) {
  HashMap < Integer, Integer > hm = new HashMap < > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( hm . containsKey ( arr [ i ] ) )
    hm . put ( arr [ i ], hm . get ( arr [ i ] ) + 1 );
    else hm . put ( arr [ i ], 1 );
  }
  int ans = 0;
  for ( Map . Entry < Integer, Integer > it : hm . entrySet ( ) ) {
    int count = it . getValue ( );
    ans += ( count * ( count - 1 ) ) / 2;
  }
  return ans;
}